src/components/color-picker/color-picker.tsx

```diff
@@ -0,0 +1,151 @@
+import { Component, h, Prop, State, Event, EventEmitter } from '@stencil/core';
+import { FormComponent } from '../form/form.types';
+
+/**
+ * This component enables you to select a swatch from out color palette, simply
+ * by clicking on it. You can then copy the css variable name of the chosen color
+ * and use it where desired.
+ *
+ * The color picker can also show you a preview of any valid color name or color value.
+ *
+ * :::note
+ * Make sure to read our [guidelines about usage of colors](/#/DesignGuidelines/color-system.md/) from our palette.
+ * :::
+ *
+ * @exampleComponent limel-example-color-picker
+ * @exampleComponent limel-example-color-picker-readonly
+ */
+
+@Component({
+    tag: 'limel-color-picker',
+    shadow: true,
+    styleUrl: 'color-picker.scss',
+})
+export class ColorPicker implements FormComponent {
+    /**
+     * Name or code of the chosen color
+     */
+    @Prop({ reflect: true })
+    public value: string;
+
+    /**
+     * The label of the input field
+     */
+    @Prop({ reflect: true })
+    public label: string;
+
+    /**
+     * Helper text of the input field
+     */
+    @Prop({ reflect: true })
+    public helperText: string;
+
+    /**
+     * Displayed as tooltips when picker is hovered.
+     */
+    @Prop({ reflect: true })
+    public tooltipLabel: string;
+
+    /**
+     * Set to `true` if a value is required
+     */
+    @Prop({ reflect: true })
+    public required: boolean;
+
+    /**
+     * Set to `true` if a value is readonly. This makes the component un-interactive.
+     */
+    @Prop({ reflect: true })
+    public readonly: boolean;
```

Comment:
One thing I want us to keep in mind is that we have made two distinct components here. One is `limel-color-picker` which is what we display to the public, and the 2nd one is `limel-color-picker-palette` which is private. But we can potentially use it in other places too. Below could be an example of such implementation:
![image](https://user-images.githubusercontent.com/35954987/144115083-cd4439c4-c274-4918-8aff-263cb0267a57.png)

Which is why we have `readonly` here, but not for `limel-color-picker-palette`. For that component, it doesn't make sense to have a `readonly` mode.